### ISSUE TITLE
Camera

### ISSUE THREAD

**Issue by hajimehoshi** [hajimehoshi/ebiten#1153]
A camera is one of the most requested feature for Ebiten. I don't have a concrete idea so far.

**Comment by KuliuTho** [comment]
Instead of passing a screen *Image directly to the draw function maybe pass a struct that has all the same functionality DrawImage etc but it will behind the scenes multiply the DrawImageOptions with camera GeoM?

Something like this 

> type ScreenRenderer struct {
	Screen *ebiten.Image
	CameraMatrix *ebiten.GeoM
}
func (renderer *ScreenRenderer) DrawImage(image *ebiten.Image, options *ebiten.DrawImageOptions) {
	matrix := renderer.CameraMatrix
	options.GeoM.Add(*matrix)
	renderer.Screen.DrawImage(image, options)
}

**Comment by hajimehoshi** [comment]
Yes, I think this represents the core idea. Thank you! Another necessary thing is clipping (https://github.com/hajimehoshi/ebiten/issues/996).

> options.GeoM.Add(*matrix)

You meant Concat? And as the order matters, the GeoM would be like this:

```
options.GeoM = renderer.CameraMatrix.Concat(options.GeoM)
```

**Comment by KuliuTho** [comment]
> Yes, I think this represents the core idea. Thank you! Another necessary thing is clipping (#996).
> 
> > options.GeoM.Add(*matrix)
> 
> You meant Concat? And as the order matters, the GeoM would be like this:
> 
> ```
> options.GeoM = renderer.CameraMatrix.Concat(options.GeoM)
> ```
> 
> EDIT: Oops, this should have been:
> 
> ```
> options.GeoM.Concat(renderer.CameraMatrix)
> ```
> 
> ?

Yes sorry I meant Concat!
